apps/confidential/examples_test.go

```diff
@@ -35,3 +36,51 @@ func ExampleNewCredFromCert_pem() {
 	}
 	fmt.Println(cred) // Simply here so cred is used, otherwise won't compile.
 }
+
+func ExampleClient_AcquireTokenByCredential() {
+	// In this case, we are getting a credential using a secret.
+	// We could also use an assertion (NewCredFromAssertion) or a certificate (NewCredFromCert) to obtain a credential.
+	var secret = "the_secret"
+	cred, err := NewCredFromSecret(secret)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByCredential: acquring token by credential %v", err)
+	}
+	client, err := New("client_id", cred)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByCredential: acquring token by credential %v", err)
+	}
+
+	ctx := context.Background()
+	var tokenScope = []string{"the_scope"}
+
+	result, err := client.AcquireTokenByCredential(ctx, tokenScope)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByCredential: acquring token by credential %v", err)
+	}
+	fmt.Println(result)
+
+}
+
+func ExampleClient_AcquireTokenByAuthCode() {
+	// In this case, we are getting a credential using a secret.
+	// We could also use an assertion (NewCredFromAssertion) or a certificate (NewCredFromCert) to obtain a credential.
+	var secret = "the_secret"
+	cred, err := NewCredFromSecret(secret)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByAuthCode: acquring token by auth code %v", err)
+	}
+	client, err := New("client_id", cred)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByAuthCode: acquring token by auth code %v", err)
+	}
+
+	ctx := context.Background()
+	var tokenScope = []string{"the_scope"}
+
+	result, err := client.AcquireTokenByAuthCode(ctx, "auth_code", "redirect_uri", tokenScope)
```

Comment:
While this shows how to invoke the API, I think it needs more documentation.  Where does `auth_code` come from?  What value do I use for `redirect_uri`?  The auth code flow is a bit complicated, so we should have clear docs on how to use it end-to-end.  If that exists elsewhere that's fine, we should then add a code comment with a link to them.

Reply:
Thank you, let me fix that.